### DISCUSSION TITLE
Wavelog - baseurl problem. (solution for nginx proxy manager)

### DISCUSSION THREAD

**Discussion by atimias** [wavelog/wavelog#1231] [Q&A] [answered]
**This is for reporting bugs with the code, if this issue is regarding setting up on your local server please use the discussions area**

**Describe the bug**
Baseurl rewrites address what makes it hard to work both by subdomain (using Nginx Proxy Manager) and also by local IP address.

**Version Information**
We need some information about your versions. You can get all Information in the Admin > Debug Menu
- OS: Debian 12
- PHP: 8.2.24
- MariaDB: 11.3
- Wavelog: 1.8.8

**Steps to Reproduce**
1. Build wavelog from official docker image.
2. Set baseurl to 192.168.1.X
3. Configure NPM proxy_host to reach 192.168.1.X:8086
4. When using subdomain.domain.tld the url is being rewritten. Baseurl has to be set to subdomain.domain.tld, but then - won't work by local IP, or breaks css and so on.

**Expected Behavior**
I'd like to make an one instance to work on private and public subdomain without tinkering.

**Screenshots**

**Desktop Information:**
- OS: Debian 12
- Browser: Chrome/Firefox
- Version: current

**Additional Context**
Maybe I am overthinking the problem. Not so great experience in Nginx Proxy Manager.
Looking forward for help.

Thank you!

**Comment by HB9HIL** [comment]
Hello @atimias 

The `baseurl` is part if the Codeigniter Framework. We unfortunately can not simpley override this. 

I'm sorry. 

If you want to reach your installation also in a local network without having access from outside you could add a local DNS override which points to the NGPM. But this is not Wavelog and not a bug.

**Reply by uiolee** [reply]
I tried setting `$config['base_url']` to `/` and I found that it works fine. 

But I don't fully understand and am not sure if this has any side effects

**Comment by HB9HIL** [comment]
I converted this issue to a general question in discussions.

**Comment by uiolee** [comment]
This is an annoying issue where the base_url is solidified in the code. I only had to use nginx reverse proxy to solve this problem

```php
# application/config/config.php
$config['base_url'] = 'http://localhost:114514/';
```

```nginx
# reverse proxy for wavelog
server {
    listen 66666;
    listen [::]:66666;
    server_name 1.example.com 2.example.com 3.example.com;

    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $server_name;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;

    index null.html;
    location ^~ / {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header REMOTE-HOST $remote_addr;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_http_version 1.1;
        add_header X-Cache $upstream_cache_status;
        add_header Cache-Control no-cache;
        proxy_ssl_server_name off;
        proxy_ssl_name $proxy_host;

        proxy_pass http://localhost:114514/;
        proxy_redirect http://localhost:114514/ /;

        proxy_set_header Accept-Encoding "";
        sub_filter_once off;
        sub_filter "http://localhost:114514/" "/";
        sub_filter_types *;
    }
}
```

**Comment by desertblade** [comment, marked as answer]
Thanks to @uiolee example, I got it working in Nginx Proxy Manager (NPM) by adding this under the Advanced tab when editing proxy host in the GUI. It uses NPM's variables so it should be an easy copy/paste.

```javascript
location / { 
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
        
    proxy_pass http://$server:$port;
    proxy_redirect http://$server:$port/ https://$server_name/;

    proxy_set_header Accept-Encoding "";
    sub_filter_once off;
    sub_filter "http://$server:$port" "https://$server_name";
    sub_filter_types *; 
}
```

If you want to redirect / to the visitor page add this in the Advanced Tab, where "home" is your station name.

```javascript
location = /{return 301 $scheme://$http_host/visitor/home;}
```

In theory you can use this to block "/user/login" and other pages.
